### PR TITLE
[Tour of Beam] Save code, Delete account & other tasks

### LINKED ISSUE #24538: [Task]: Save user code [ToB] [Frontend]
### What needs to happen?

Automatically save snippets after a user edits unit example/challenge code. 

### Issue Priority

Priority: 3

### Issue Component

Component: beam-playground

### LINKED ISSUE #25116: [Tour of Beam] [Task] Improve user experience when solving ToB challenges
### What needs to happen?
Create learning content and challenge for "Tour of Beam"::"Basic Transforms" module consisting of following unit:

We need to allow user to switch between challenge assignment and challenge solution and see what is currently displayed -  challenge or solution. Let's change button caption Solution/Assignment. Also we need to show a "we encourage you to solve all challenges by yourself" warning before showing solution. Let's use following text for warning:

""Before revealing the solution, try solving the challenge on your own. Remember, the more you practice, the better you will become. Give it a shot and see how far you can get.""

### Issue Priority
Priority: 3

### Component
tour-of-beam

### LINKED ISSUE #25255: [Tour of Beam] [Task] delete user progress [Frontend]
### What needs to happen?
Need to add frontend support for delete user progress feature for Tour of Beam
Backend part is already implemented. See #24659

Issue Priority
Priority: 3

### LINKED ISSUE #25283: [Tour of Beam] [Task] Save user code for anonymous user
### What needs to happen?
Need to save user code for anonymous users in ToB frontend

### Issue Priority
Priority: 3

### Components:
tour-of-beam

### LINKED ISSUE #25284: [Tour of Beam] [Task] Support Github authentication for ToB (Frontend)
### What needs to happen?
Need to add support for Firebase github authentication for ToB frontend

### Issue Priority
Priority: 3

### Components:
tour-of-beam

### LINKED ISSUE #25285: [Tour of Beam] [Task] Save user code for multifile examples
### What needs to happen?
Need to update saving user code feature for multifile example

### Issue Priority
Priority: 3

### Components:
tour-of-beam

### LINKED ISSUE #25468: [Tour of Beam] [Task] Load example metadata in ToB to use precompiled objects
### What needs to happen?
Tour of beam should load not only snippets but also all example metadata to show precompiled objects if any

### Issue Priority
Priority: 3

### Components:
tour-of-beam

### LINKED ISSUE #25529: [Tour of Beam] [Task] Simplify scrolling through examples in code editor
### What needs to happen?
Scrolling through examples with long line is complicated due to scrollbar  is hard to find and not rendered until you click.
Need to improve user experience scrolling through ToB examples.

### Issue Priority
Priority: 3

### Components:
tour-of-beam, beam-playground

### LINKED ISSUE #25575: [Tour of beam] [Task] Scrolling through learning materials is broken when hint is present
### What needs to happen:

Looks like scrolling through learning materials is broken when Hint button is displayed. Need to fix scrolling.

### Issue priority:
P3

### Components:
tour-of-beam

### LINKED ISSUE #26025: [Tour of Beam] [Task] Show Beam version information in ToB
### What needs to be done:
Need to use GetMetadata PG endpoint to show beam version 

### Priority:
P3

### Components:
tour-of-beam

### PR DESCRIPTION
**Pull request by nausharipov**
Resolves #25283
Resolves #24538
Resolves #25575
Resolves #25529
Resolves #25255
Resolves #25116
Resolves #25284
Resolves #25285
Resolves #26025
Addresses #25468

All changes have been approved by @alexeyinkin in https://github.com/apache/beam/pull/25347, https://github.com/apache/beam/pull/25298, https://github.com/akvelon/beam/pull/441, and https://github.com/akvelon/beam/pull/446.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by nausharipov** [comment]
R: @damondouglas

**Review by damccorm — COMMENTED** [review]
Just had a small couple comments, otherwise LGTM functionally and from the code review I was able to do

File: learning/tour-of-beam/frontend/lib/auth/notifier.dart [hunk 1]
```diff
@@ -21,9 +21,16 @@ import 'dart:async';
...
+import 'package:playground_components/playground_components.dart';
+
+import '../cache/unit_progress.dart';
+import '../repositories/client/client.dart';
 
 class AuthNotifier extends ChangeNotifier {
-  AuthNotifier() {
+  final TobClient client;
+
+  AuthNotifier({required this.client}) {
```
[the remark this answers is not in the document]
Comment (nausharipov) [reply learning/tour-of-beam/frontend/lib/auth/notifier.dart]: Deleted.

File: learning/tour-of-beam/frontend/lib/config.dart [hunk 2]
```diff
@@ -18,8 +18,20 @@
...
 
-const _cloudFunctionsProjectRegion = 'us-central1';
-const _cloudFunctionsProjectId = 'tour-of-beam-2';
+const _cloudFunctionsProjectRegion = 'us-east1';
+const _cloudFunctionsProjectId = 'river-key-378010';
 const cloudFunctionsBaseUrl = 'https://'
     '$_cloudFunctionsProjectRegion-$_cloudFunctionsProjectId'
     '.cloudfunctions.net';
+
+// Copied from Playground's config.g.dart
```
[the remark this answers is not in the document]
Comment (nausharipov) [reply learning/tour-of-beam/frontend/lib/config.dart]: No. Deleted.

**Pull request merged**